jormungandr-lib/src/interfaces/block0_configuration/initial_fragment.rs

```diff
@@ -43,68 +51,87 @@ pub struct LegacyUTxO {
 
 #[derive(Debug, Error)]
 pub enum Error {
-    #[error("first message of block 0 is not initial")]
-    FirstBlock0MessageNotInit,
-    #[error("non-first message of block 0 has unexpected type")]
-    Block0MessageUnexpected,
     #[error("initial UTXO has input")]
     InitUtxoHasInput,
+    #[error("non-first message of block 0 has unexpected type")]
+    Block0MessageUnexpected,
 }
 
-pub fn try_initials_vec_from_messages<'a>(
-    messages: impl Iterator<Item = &'a Fragment>,
-) -> Result<Vec<Initial>, Error> {
-    let mut inits = Vec::new();
-    for message in messages {
-        match message {
-            Fragment::Transaction(tx) => try_extend_inits_with_tx(&mut inits, &tx.as_slice())?,
-            Fragment::OldUtxoDeclaration(utxo) => extend_inits_with_legacy_utxo(&mut inits, utxo),
-            Fragment::PoolRegistration(tx) => {
-                let tx = tx.as_slice();
-                let cert = tx.payload().into_payload();
-                let auth = tx.payload_auth().into_payload_auth();
-                let cert = certificate::SignedCertificate::PoolRegistration(cert, auth);
-                inits.push(Initial::Cert(cert.into()))
-            }
-            Fragment::StakeDelegation(tx) => {
-                let tx = tx.as_slice();
-                let cert = tx.payload().into_payload();
-                let auth = tx.payload_auth().into_payload_auth();
-                let cert = certificate::SignedCertificate::StakeDelegation(cert, auth);
-                inits.push(Initial::Cert(cert.into()))
-            }
-            Fragment::VotePlan(tx) => {
-                let tx = tx.as_slice();
-                let cert = tx.payload().into_payload();
-                // the pattern match here is to make sure we are actually expecting the `()`
-                // and that if it changes the compiler will detect it and tell us about the
-                // change so we are reminded of a breaking change
-                let auth = tx.payload_auth().into_payload_auth();
-                let cert = certificate::SignedCertificate::VotePlan(cert, auth);
-                inits.push(Initial::Cert(cert.into()))
-            }
-            _ => return Err(Error::Block0MessageUnexpected),
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Destination {
+    address: Identifier,
```

Comment:
Could this be an actual address and not an `Identifier` for consistency with funds? IIRC it should not be hard to do the conversion.

Reply:
I am not sure that it is possible, because we will have problems to convert `Address` into `Identifier` in both directions, which is needed there.
`Address` -> `Identifier` problems:
- `Address` type can be `Group` or `Multisig` or `Script` which is hardly to somehow convert or determine a public key of the receiver which is needed for the MintToken fragment.

`Identifier ` -> `Address` problems:
- need to determine `DISCRIMINATION` type of the address which I am not sure that is possible at this place (but maybe I am mistaken, need to investigate)

Reply:
For the first issue we can restrict that you have to use only `Account` or `Single` type of `Address`.

Reply:
It is possible, I have done 😊